crates/iota-data-ingestion-core/src/executor.rs

```diff
@@ -126,8 +127,8 @@ pub async fn setup_single_workflow<W: Worker + 'static>(
             Some(remote_store_url),
             vec![],
             reader_options.unwrap_or_default(),
-            exit_receiver,
+            token.child_token(),
```

Comment:
Nit: Do we need a child token here, instead of simple clone? Is there a caller that will cancel the child token only, so that we require this granularity of cancellation?

Referring to this specification: https://docs.rs/tokio-util/latest/tokio_util/sync/struct.CancellationToken.html#method.child_token

Reply:
You're right, thank you, [fixed](https://github.com/iotaledger/iota/pull/4638/commits/5eee78cc3f28983a37582f7473d828e1db9a186f#diff-fc30500a6b8c33380223c735cb9d6786afa2cc39301199f614993173f3acdb85R130)